### ISSUE TITLE
Using --createstub with TypeVar variables loses type information

### ISSUE THREAD

**Issue by SandyChapman** [microsoft/pyright#2961]
So I don't believe this is a bug since the behaviour is consistent with other variables defined at the global scope, however, I feel a special case should be added for TypeVars.

For example: consider the following Python:

```python
from typing import Generic, TypeVar

T = TypeVar('T')

class Y(Generic[T]):

    def __init__(self, y: T):
        self.y = y 
```

When running `pyright --createstub` for this module, we get the following output:

```python
"""
This typestub file was generated by pyright.
"""

from typing import Generic, TypeVar

T = ...
class Y(Generic[T]):
    def __init__(self, y: T) -> None:
        ...
```

In this output, Pyright shows an error on the Generic param `T`:

```
Type argument for "Generic" must be a type variable
```

Additionally, any code that tries to specific the generic type when using the class Y elicits an error from Pyright.

```python
from main import Y

y: Y[int] = Y(5) # Expected no type arguments for class "Y"
```

Correcting `T = ...` to `T = TypeVar('T')` in the stub resolves both errors. I understand that the generated stubs are intended to be edited, however, I think in this case, pyright should simply generate the `TypeVar` variables correctly in the first place. Lastly, this is helpful behaviour as we are generating some stubs for large libraries that need to be updated periodically. Any additional edits to the typestubs as generated requires additional overhead when updating stubs to a new version.

Thanks for your consideration.

**Comment by erictraut** [comment]
Yes, this is a reasonable request. I think it hasn't come up before because code bases that contain TypeVar definitions generally already contain inlined type annotations, so it's not typical to generate stubs from such code.

Out of curiosity, are you generating stubs from fully-typed code? If so, where do you intend to use these stubs? I ask because we've generally made the assumption that modules with inlined types are easier to use and maintain than separate type stub files.

**Comment by SandyChapman** [comment]
The specific instance we're hitting is with apache_beam's Python SDK here: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/pvalue.py#L140

We're trying to annotate the `PCollection`'s generic parameter and receiving an error that it's not parameterizable because our generated stubs have `T = ...`. It's odd behaviour as well, because the Generic class can be fully typed and still hit this problem. I was actually a little surprised it hasn't been mentioned before.

**Comment by erictraut** [comment]
I've added special-case handling for TypeVar, TypeVarTuple, ParamSpec and NewType definitions. These are now all emitted into the type stub. This change will be included in the next release.

**Comment by SandyChapman** [comment]
Thanks again for the fast turnaround!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.217, which I just published. It will also be included in the next release of pylance.
